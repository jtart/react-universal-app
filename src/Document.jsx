import React from 'react';
import Root from './Root.jsx';

const Document = ({ assets, content, data, helmet, styles }) => {
  const htmlAttrs = helmet.htmlAttributes.toComponent();
  const bodyAttrs = helmet.bodyAttributes.toComponent();

  return (
    <html {...htmlAttrs}>
      <head>
        {helmet.title.toComponent()}
        {helmet.meta.toComponent()}
        {helmet.link.toComponent()}
        {styles}
      </head>
      <body {...bodyAttrs}>
        <Root content={content} data={data} assets={assets} />
      </body>
    </html>
  );
};

export default Document;
