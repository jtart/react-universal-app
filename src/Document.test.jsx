import React from 'react';
import Document from './Document';
import Helmet from 'react-helmet';

Helmet.renderStatic = jest.fn().mockReturnValue({
  htmlAttributes: {
    toComponent: jest.fn().mockReturnValue({ lang: 'en_GB' }),
  },
  title: {
    toComponent: jest.fn().mockReturnValue(<title />),
  },
  meta: {
    toComponent: jest.fn().mockReturnValue(<meta />),
  },
  link: {
    toComponent: jest
      .fn()
      .mockReturnValue([<link key="l-1" />, <link key="l-2" />]),
  },
  bodyAttributes: {
    toComponent: jest.fn().mockReturnValue({ dir: 'ltr' }),
  },
});

describe('Document', () => {
  it('should render correctly', () => {
    const doc = (
      <Document
        additionalHeadProps={[
          <style key="s-1" data-test-styles-one />,
          <style key="s-2" data-test-styles-two />,
        ]}
        appHTML={<h1>This is some html</h1>}
        data={{ data: 'This is some data!' }}
        scripts={['https://example.com/one.js', 'https://example.com/two.js']}
      />
    );
    const wrapper = shallow(doc);
    expect(wrapper).toMatchSnapshot();
  });
});
