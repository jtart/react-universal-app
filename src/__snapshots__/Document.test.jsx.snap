// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document should render correctly 1`] = `
<Document
  additional={
    Array [
      <style
        data-test-styles-one={true}
      />,
      <style
        data-test-styles-two={true}
      />,
    ]
  }
  data={
    Object {
      "data": "This is some data!",
    }
  }
  html={
    <h1>
      This is some html
    </h1>
  }
  scripts={
    Array [
      "https://example.com/one.js",
      "https://example.com/two.js",
    ]
  }
>
  <html
    lang="en_GB"
  >
    <head>
      <title />
      <meta />
      <link
        key="l-1"
      />
      <link
        key="l-2"
      />
      <style
        data-test-styles-one={true}
        key="s-1"
      />
      <style
        data-test-styles-two={true}
        key="s-2"
      />
    </head>
    <body
      dir="ltr"
    >
      <Root
        html={
          <h1>
            This is some html
          </h1>
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": <h1>
                This is some html
              </h1>,
            }
          }
          id="__uni__"
        />
      </Root>
      <ServerData
        data={
          Object {
            "data": "This is some data!",
          }
        }
        id="__UNI_DATA__"
      >
        <script
          dangerouslySetInnerHTML={
            Object {
              "__html": "{\\"data\\":\\"This is some data!\\"}",
            }
          }
          id="__UNI_DATA__"
          type="application/json"
        />
      </ServerData>
      <Scripts
        scripts={
          Array [
            "https://example.com/one.js",
            "https://example.com/two.js",
          ]
        }
      >
        <script
          defer={true}
          key="script-https://example.com/one.js"
          src="https://example.com/one.js"
        />
        <script
          defer={true}
          key="script-https://example.com/two.js"
          src="https://example.com/two.js"
        />
      </Scripts>
    </body>
  </html>
</Document>
`;
